src/pages/SudoPrompt.vue

```diff
@@ -0,0 +1,83 @@
+<!--
+  - This is a modal dialog displayed before we ask the user for a sudo password
+  - to explain why we're asking for it.
+  -->
+
+<template>
+  <div class="contents">
+    <h2>Administrative Access Required</h2>
+    <p>
+      Rancher Desktop requires administrative credentials ("sudo access") in
+      order to provide a better experience.  We would like to have access for
+      the following:
+    </p>
+    <ul>
+      <li v-for="item in explanations" :key="item" class="monospace" v-text="item" />
+    </ul>
+    <p>
+      We will display the actual prompt once this window is closed.  Cancelling
+      the password prompt will cause Rancher Desktop to run in reduced
+      functionality mode, but it should still start.
+    </p>
+    <checkbox
+      id="suppress"
+      v-model="suppress"
+      label="Always run without administrative access"
+    />
+    <button ref="accept" class="role-primary" @click="close">
+      OK
+    </button>
+  </div>
+</template>
+
+<script lang="ts">
+import { ipcRenderer } from 'electron';
+import Vue from 'vue';
+import Checkbox from '@/components/form/Checkbox.vue';
+
+export default Vue.extend({
+  components: { Checkbox },
+  layout:     'dialog',
+  data() {
+    return {
+      explanations:    [] as string[],
+      suppress:        false,
+    };
+  },
+  mounted() {
+    ipcRenderer.on('dialog/populate', (event, explanations) => {
+      this.explanations = explanations;
+    });
+    window.addEventListener('close', () => {
+      ipcRenderer.send('sudo-prompt/closed', this.suppress);
+    });
+    (this.$refs.accept as HTMLButtonElement)?.focus();
+  },
+  methods: {
+    close() {
+      // Manually send the result, because we won't get an event here.
+      ipcRenderer.send('sudo-prompt/closed', this.suppress);
+      window.close();
```

Comment:
There are two ways for the window to be closed:
- The user presses the button.
- The user uses OS-level window controls.

In the first case, with the suggested change, the event flow would be:
- Button is pressed
- We send `sudo-prompt/closed` to the main process
- The main process closes the window.

In the second case, with the suggested change, the event flow would be:
- The window is closed.
- The `close` event fires; there are no listeners.
- … we need to make the main process wait for window close to get the results?

Ultimately, the issue is that we still need to listen for the window closing, because we can't limit the ways the user closes the window.

Reply:
Makes sense to me, thanks for the explanation.

Note for the future, we might want to consider how we expect closing windows (more importantly, dialogs) to behave. It seems like we're treating closing the window as an implicit `OK` and I don't think that would be expected in all scenarios..